submissions/ASTkach/js-dom/js/script.js

```diff
@@ -0,0 +1,495 @@
+                        <p class="list__info">${tempItem.name}: <span class="list__value">${tempItem.value} </span><sup>o</sup>C</p>
+                    </li>`;
+                });
+
+                planetTemplateTemp += planetTemplateTempContent;
+            }
+
+            let planetTemplateMoons = "";
+
+            if (planetMoons) {
+                let planetTemplateMoonsContent = "";
+
+                planetMoons.forEach((moonsItem) => {
+                    planetTemplateMoonsContent += `
+                    <li class="list__item">
+                        <p class="list__info list__info--flax">
+                            ${moonsItem.name}: <span class="list__value"> <img src="img/icons/moon.png" alt="" class="list__icon" width="100" height="104"/></span><span> * ${moonsItem.value}</span>
+                        </p>
+                    </li>`;
+                });
+
+                planetTemplateMoons += planetTemplateMoonsContent;
+            }
+
+            let planetTemplateDistance = "";
+
+            if (planetDistance) {
+                let planetTemplateDistanceContent = "";
+
+                planetDistance.forEach((distanceItem) => {
+                    planetTemplateDistanceContent += `
+                    <li class="list__item">
+                        <p class="list__info">${distanceItem.name}: <span class="list__value">${distanceItem.value} </span>${distanceItem.unit}</p>
+                    </li>`;
+                });
+
+                planetTemplateDistance += planetTemplateDistanceContent;
+            }
+
+            let planetTemplate = "";
+            planetTemplate += planetTemplateStart;
+            planetTemplate += planetTemplateTextContent;
+            planetTemplate += planetTemplateImage;
+            planetTemplate += planetTemplateListStart;
+            planetTemplate += planetTemplateListItems;
+            planetTemplate += planetTemplateTemp;
+            planetTemplate += planetTemplateMoons;
+            planetTemplate += planetTemplateDistance;
+            planetTemplate += planetTemplateListEnd;
+            planetTemplate += planetTemplateEnd;
+
+            mainBlock.innerHTML = planetTemplate;
+        }
+    });
+}
+
+// create homepage
+getCurrentPlanet(0);
+
+const iconMenu = document.querySelector(".nav__icon");
+const listMenu = document.querySelector(".nav__body");
+
+// track the "click" event on "burger menu"
+iconMenu.addEventListener("click", function () {
+    document.body.classList.toggle("_lock");
+    iconMenu.classList.toggle("_opened");
+    listMenu.classList.toggle("_opened");
+});
+
+// track the presence of the "_active" class for navigation elements
+// track the "click" event on them
+let navButton = document.getElementsByClassName("nav__button");
+let activeClass = document.getElementsByClassName("_active");
+for (i = 0; navButton.length > i; i++) {
+    navButton[i].onclick = function () {
+        let currentActive = activeClass[0];
```

Comment:
Read about [event delegation](https://javascript.info/event-delegation)

Reply:
I tried to do it with event delegation. but my picture does not want to respond to a click in any way, unlike a paragraph. I even just removed all the styles from the page, but even that did not help

Reply:
Find me in telegram '@madmaxWMFU'